Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\core\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /login/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\core\views.py", line 96, in login_user
    return render(request, 'core/login.html')
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'custom_login' not found. 'custom_login' is not a valid view function or pattern name.
"POST /login/ HTTP/1.1" 500 151781
Internal Server Error: /login/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\core\views.py", line 96, in login_user
    return render(request, 'core/login.html')
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'login_user' not found. 'login_user' is not a valid view function or pattern name.
"POST /login/ HTTP/1.1" 500 151729
"POST /login/ HTTP/1.1" 200 5754
"GET /signup/ HTTP/1.1" 200 7485
"POST /signup/ HTTP/1.1" 200 8164
"POST /signup/ HTTP/1.1" 200 7880
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\core\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\core\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\core\views.py changed, reloading.
Watching for file changes with StatReloader
"POST /signup/ HTTP/1.1" 200 8160
"POST /signup/ HTTP/1.1" 200 8180
Internal Server Error: /signup/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\core\views.py", line 16, in signup
    logger.info(f"Signup attempt with data: {request.POST}")
    ^^^^^^
NameError: name 'logger' is not defined
"POST /signup/ HTTP/1.1" 500 69933
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\core\views.py changed, reloading.
Watching for file changes with StatReloader
"POST /signup/ HTTP/1.1" 200 8157
"GET /login/ HTTP/1.1" 200 5408
"POST /login/ HTTP/1.1" 302 0
"GET / HTTP/1.1" 200 4206
"GET /logout/ HTTP/1.1" 302 0
"GET / HTTP/1.1" 200 4115
"GET /login/ HTTP/1.1" 200 5747
"POST /login/ HTTP/1.1" 302 0
"GET / HTTP/1.1" 200 4206
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\core\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\Enfold_chat\settings.py changed, reloading.
Watching for file changes with StatReloader
"GET /login/ HTTP/1.1" 200 5838
Internal Server Error: /login/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\core\views.py", line 64, in login_user
    return redirect('rooms')
           ^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\shortcuts.py", line 49, in redirect
    return redirect_class(resolve_url(to, *args, **kwargs))
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\shortcuts.py", line 180, in resolve_url
    return reverse(to, args=args, kwargs=kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'rooms' not found. 'rooms' is not a valid view function or pattern name.
"POST /login/ HTTP/1.1" 500 88977
"GET /login/ HTTP/1.1" 200 5838
"GET / HTTP/1.1" 200 4206
"GET / HTTP/1.1" 200 4206
"GET /login/ HTTP/1.1" 200 5499
Internal Server Error: /login/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\core\views.py", line 64, in login_user
    return redirect('rooms')
           ^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\shortcuts.py", line 49, in redirect
    return redirect_class(resolve_url(to, *args, **kwargs))
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\shortcuts.py", line 180, in resolve_url
    return reverse(to, args=args, kwargs=kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'rooms' not found. 'rooms' is not a valid view function or pattern name.
"POST /login/ HTTP/1.1" 500 88977
"GET /login/ HTTP/1.1" 200 5838
"GET / HTTP/1.1" 200 4206
"GET /logout/ HTTP/1.1" 302 0
"GET / HTTP/1.1" 200 4115
"GET /login/ HTTP/1.1" 200 5408
Internal Server Error: /login/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\core\views.py", line 64, in login_user
    return redirect('rooms')
           ^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\shortcuts.py", line 49, in redirect
    return redirect_class(resolve_url(to, *args, **kwargs))
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\shortcuts.py", line 180, in resolve_url
    return reverse(to, args=args, kwargs=kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'rooms' not found. 'rooms' is not a valid view function or pattern name.
"POST /login/ HTTP/1.1" 500 88828
"GET /login/ HTTP/1.1" 200 5747
"GET / HTTP/1.1" 200 4115
"GET /signup/ HTTP/1.1" 200 7485
"GET / HTTP/1.1" 200 4115
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 4115
Not Found: /favicon.ico
"GET /favicon.ico HTTP/1.1" 404 3227
"GET /login/ HTTP/1.1" 200 5408
Internal Server Error: /login/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\core\views.py", line 64, in login_user
    return redirect('rooms')
           ^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\shortcuts.py", line 49, in redirect
    return redirect_class(resolve_url(to, *args, **kwargs))
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\shortcuts.py", line 180, in resolve_url
    return reverse(to, args=args, kwargs=kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'rooms' not found. 'rooms' is not a valid view function or pattern name.
"POST /login/ HTTP/1.1" 500 88523
"GET /login/ HTTP/1.1" 200 5747
"GET / HTTP/1.1" 200 4115
"GET /login/ HTTP/1.1" 200 5408
"GET / HTTP/1.1" 200 4115
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 4115
"GET / HTTP/1.1" 200 4115
Not Found: /favicon.ico
"GET /favicon.ico HTTP/1.1" 404 3227
"GET /admin HTTP/1.1" 301 0
"GET /admin/ HTTP/1.1" 302 0
"GET /admin/login/?next=/admin/ HTTP/1.1" 200 4160
"GET /static/admin/css/responsive.css HTTP/1.1" 200 17972
"GET /static/admin/css/base.css HTTP/1.1" 200 22092
"GET /static/admin/css/login.css HTTP/1.1" 200 951
"GET /static/admin/css/nav_sidebar.css HTTP/1.1" 200 2810
"GET /static/admin/css/dark_mode.css HTTP/1.1" 200 2804
"GET /static/admin/js/theme.js HTTP/1.1" 200 1653
"GET /static/admin/js/nav_sidebar.js HTTP/1.1" 200 3063
- Broken pipe from ('127.0.0.1', 64455)
"POST /admin/login/?next=/admin/ HTTP/1.1" 302 0
"GET /admin/ HTTP/1.1" 200 5852
"GET /static/admin/css/dashboard.css HTTP/1.1" 200 441
"GET /static/admin/img/icon-addlink.svg HTTP/1.1" 200 331
"GET /static/admin/img/icon-changelink.svg HTTP/1.1" 200 380
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\admin.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\admin.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\admin.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\admin.py changed, reloading.
Watching for file changes with StatReloader
"GET /admin/ HTTP/1.1" 200 6633
"GET /admin/room/room/add/ HTTP/1.1" 200 9126
"GET /admin/jsi18n/ HTTP/1.1" 200 3342
"GET /static/admin/css/forms.css HTTP/1.1" 200 8794
"GET /static/admin/js/jquery.init.js HTTP/1.1" 200 347
"GET /static/admin/js/urlify.js HTTP/1.1" 200 7887
"GET /static/admin/css/widgets.css HTTP/1.1" 200 11564
"GET /static/admin/js/prepopulate.js HTTP/1.1" 200 1531
"GET /static/admin/js/core.js HTTP/1.1" 200 6208
"GET /static/admin/js/actions.js HTTP/1.1" 200 8076
"GET /static/admin/js/admin/RelatedObjectLookups.js HTTP/1.1" 200 9097
"GET /static/admin/js/prepopulate_init.js HTTP/1.1" 200 586
"GET /static/admin/js/vendor/jquery/jquery.js HTTP/1.1" 200 285314
"GET /static/admin/js/change_form.js HTTP/1.1" 200 606
"GET /static/admin/js/vendor/xregexp/xregexp.js HTTP/1.1" 200 325171
"POST /admin/room/room/add/ HTTP/1.1" 302 0
"GET /admin/room/room/ HTTP/1.1" 200 9230
"GET /admin/jsi18n/ HTTP/1.1" 200 3342
"GET /static/admin/css/changelists.css HTTP/1.1" 200 6878
"GET /static/admin/js/filters.js HTTP/1.1" 200 978
"GET /static/admin/img/icon-yes.svg HTTP/1.1" 200 436
"GET /static/admin/img/tooltag-add.svg HTTP/1.1" 200 331
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\Enfold_chat\urls.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\Enfold_chat\urls.py changed, reloading.
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 4206
"GET /logout/ HTTP/1.1" 302 0
"GET / HTTP/1.1" 200 4115
"GET /login/ HTTP/1.1" 200 5408
"POST /login/ HTTP/1.1" 302 0
Internal Server Error: /rooms/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 8, in rooms
    rooms = room.objects.all()
            ^^^^
NameError: name 'room' is not defined. Did you mean: 'Room'?
"GET /rooms/ HTTP/1.1" 500 72966
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /rooms/ HTTP/1.1" 200 3200
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\models.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\admin.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\admin.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
"GET /rooms/ HTTP/1.1" 200 3911
"GET /logout/ HTTP/1.1" 302 0
"GET / HTTP/1.1" 200 4115
"GET /login/ HTTP/1.1" 200 5747
"POST /login/ HTTP/1.1" 302 0
"GET /rooms/ HTTP/1.1" 200 3911
"GET /rooms/ HTTP/1.1" 200 3911
"GET /logout/ HTTP/1.1" 302 0
"GET / HTTP/1.1" 200 4115
"GET /login/ HTTP/1.1" 200 5747
"POST /login/ HTTP/1.1" 302 0
"GET /rooms/ HTTP/1.1" 200 3911
"GET /rooms/ HTTP/1.1" 200 3911
"GET /admin/room/room/ HTTP/1.1" 302 0
"GET /admin/login/?next=/admin/room/room/ HTTP/1.1" 200 3816
"POST /admin/login/?next=/admin/room/room/ HTTP/1.1" 302 0
Not Found: /admin/room/room/
"GET /admin/room/room/ HTTP/1.1" 404 6850
"GET /admin/ HTTP/1.1" 200 7005
"GET /admin/room/chatroom/ HTTP/1.1" 200 7802
"GET /admin/jsi18n/ HTTP/1.1" 200 3342
"GET /admin/room/chatroom/add/ HTTP/1.1" 200 9852
"GET /admin/jsi18n/ HTTP/1.1" 200 3342
"POST /admin/room/chatroom/add/ HTTP/1.1" 302 0
"GET /admin/room/chatroom/ HTTP/1.1" 200 9323
"GET /admin/jsi18n/ HTTP/1.1" 200 3342
"GET /rooms/ HTTP/1.1" 200 4199
"GET /rooms/ HTTP/1.1" 200 4302
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\urls.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\urls.py changed, reloading.
Watching for file changes with StatReloader
"GET /rooms/ HTTP/1.1" 200 4303
Internal Server Error: /rooms/Familly/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 14, in room_detail
    return render(request, 'room/room_detail.html', {'chat_room': chat_room})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'send_message' not found. 'send_message' is not a valid view function or pattern name.
"GET /rooms/Familly/ HTTP/1.1" 500 157946
"GET /rooms/Familly/ HTTP/1.1" 200 4562
"GET /rooms/Familly/? HTTP/1.1" 200 4562
"GET /rooms/ HTTP/1.1" 200 4303
"GET /rooms/Familly/ HTTP/1.1" 200 4562
"GET /rooms/Familly/ HTTP/1.1" 200 4562
"GET /rooms/Familly/ HTTP/1.1" 200 4562
"GET /rooms/ HTTP/1.1" 200 4303
"GET /rooms/Familly/ HTTP/1.1" 200 4562
"GET /logout/ HTTP/1.1" 302 0
"GET / HTTP/1.1" 200 4115
"GET /login/ HTTP/1.1" 200 5408
"POST /login/ HTTP/1.1" 302 0
"GET /rooms/ HTTP/1.1" 200 4303
"GET /rooms/Familly/ HTTP/1.1" 200 4562
"GET /rooms/Familly/ HTTP/1.1" 200 4562
"GET /rooms/Familly/ HTTP/1.1" 200 4562
Internal Server Error: /rooms/Familly/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 14, in room_detail
    return render(request, 'room/room_detail.html', {'chat_room': chat_room})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\backends\django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader_tags.py", line 295, in do_extends
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader_tags.py", line 234, in do_block
    nodelist = parser.parse(("endblock",))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 489, in parse
    raise self.error(token, e)
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 487, in parse
    filter_expression = self.compile_filter(token.contents)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 605, in compile_filter
    return FilterExpression(token, self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 706, in __init__
    raise TemplateSyntaxError(
django.template.exceptions.TemplateSyntaxError: Could not parse the remainder: '%chat_rooms.name %' from '%chat_rooms.name %'
"GET /rooms/Familly/ HTTP/1.1" 500 186654
Internal Server Error: /rooms/Familly/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 14, in room_detail
    return render(request, 'room/room_detail.html', {'chat_room': chat_room})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader.py", line 15, in get_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\backends\django.py", line 79, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loaders\cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loaders\base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 196, in compile_nodelist
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader_tags.py", line 295, in do_extends
    nodelist = parser.parse()
               ^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 518, in parse
    raise self.error(token, e)
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 516, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader_tags.py", line 234, in do_block
    nodelist = parser.parse(("endblock",))
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 489, in parse
    raise self.error(token, e)
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 487, in parse
    filter_expression = self.compile_filter(token.contents)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 605, in compile_filter
    return FilterExpression(token, self)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 706, in __init__
    raise TemplateSyntaxError(
django.template.exceptions.TemplateSyntaxError: Could not parse the remainder: '%chat_room.name %' from '%chat_room.name %'
"GET /rooms/Familly/ HTTP/1.1" 500 186632
"GET /rooms/Familly/ HTTP/1.1" 200 4580
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\Enfold_chat\settings.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\Enfold_chat\settings.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /rooms/Familly/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 14, in room_detail
    return render(request, 'room/room_detail.html', {'chat_room': chat_room})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'send_message' not found. 'send_message' is not a valid view function or pattern name.
"GET /rooms/Familly/ HTTP/1.1" 500 158440
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\models.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\models.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\models.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\models.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 4206
"GET /login/ HTTP/1.1" 200 5499
"POST /login/ HTTP/1.1" 302 0
"GET /rooms/ HTTP/1.1" 200 4303
Internal Server Error: /rooms/Familly/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 14, in room_detail
    room = get_object_or_404(ChatRoom, slug=slug)
           ^^^^^^^^^^^^^^^^^
NameError: name 'get_object_or_404' is not defined
"GET /rooms/Familly/ HTTP/1.1" 500 79392
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /rooms/Familly/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 24, in room_detail
    return render(request, 'core/room_detail.html', {'room': room, 'messages': messages})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
django.template.exceptions.TemplateDoesNotExist: core/room_detail.html
"GET /rooms/Familly/ HTTP/1.1" 500 94795
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /rooms/Familly/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 24, in room_detail
    return render(request, 'room/room_detail.html', {'room': room, 'messages': messages})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 1008, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\base.py", line 969, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\urls\base.py", line 88, in reverse
    return resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'send_message' not found. 'send_message' is not a valid view function or pattern name.
"GET /rooms/Familly/ HTTP/1.1" 500 164400
"GET /rooms/Familly/ HTTP/1.1" 200 4739
Internal Server Error: /rooms/Familly/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 20, in room_detail
    Message.objects.create(room=room, user=request.user, content=content)
    ^^^^^^^
NameError: name 'Message' is not defined
"POST /rooms/Familly/ HTTP/1.1" 500 80316
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /rooms/Familly/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 21, in room_detail
    return redirect('room_detail', slug=room.slug)  # Redirect to the same room after sending
           ^^^^^^^^
NameError: name 'redirect' is not defined
"POST /rooms/Familly/ HTTP/1.1" 500 80203
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
"POST /rooms/Familly/ HTTP/1.1" 302 0
"GET /rooms/Familly/ HTTP/1.1" 200 5152
"POST /rooms/Familly/ HTTP/1.1" 302 0
"GET /rooms/Familly/ HTTP/1.1" 200 5420
"GET /rooms/ HTTP/1.1" 200 4303
"GET /rooms/Familly/ HTTP/1.1" 200 5420
"GET /rooms/ HTTP/1.1" 200 4303
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 4214
"GET / HTTP/1.1" 200 4123
"GET /login/ HTTP/1.1" 200 5846
"POST /login/ HTTP/1.1" 302 0
"GET /rooms/ HTTP/1.1" 200 4311
"GET /login/ HTTP/1.1" 200 5416
"POST /login/ HTTP/1.1" 200 5780
"GET /admin/room/chatroom/ HTTP/1.1" 302 0
"GET /admin/login/?next=/admin/room/chatroom/ HTTP/1.1" 200 3824
"POST /admin/login/?next=/admin/room/chatroom/ HTTP/1.1" 302 0
"GET /admin/room/chatroom/ HTTP/1.1" 200 9083
"GET /admin/jsi18n/ HTTP/1.1" 200 3342
"GET /admin/auth/user/ HTTP/1.1" 200 14394
"GET /admin/jsi18n/ HTTP/1.1" 200 3342
"GET /static/admin/img/search.svg HTTP/1.1" 200 458
"GET /static/admin/img/icon-no.svg HTTP/1.1" 200 560
"GET /static/admin/img/sorting-icons.svg HTTP/1.1" 200 1097
"GET /static/admin/img/icon-viewlink.svg HTTP/1.1" 200 581
"GET /admin/auth/user/4/change/ HTTP/1.1" 200 22597
"GET /admin/jsi18n/ HTTP/1.1" 200 3342
"GET /static/admin/js/SelectFilter2.js HTTP/1.1" 200 15502
"GET /static/admin/js/admin/DateTimeShortcuts.js HTTP/1.1" 200 19319
"GET /static/admin/js/SelectBox.js HTTP/1.1" 200 4530
"GET /static/admin/js/calendar.js HTTP/1.1" 200 9141
"GET /static/admin/img/icon-clock.svg HTTP/1.1" 200 677
"GET /static/admin/img/icon-calendar.svg HTTP/1.1" 200 1086
"GET /static/admin/img/selector-icons.svg HTTP/1.1" 200 3291
"GET /admin/auth/user/4/password/ HTTP/1.1" 200 9311
"GET /static/admin/css/unusable_password_field.css HTTP/1.1" 200 663
"GET /static/admin/js/unusable_password_field.js HTTP/1.1" 200 1480
"POST /admin/auth/user/4/password/ HTTP/1.1" 200 9585
"POST /admin/auth/user/4/password/ HTTP/1.1" 302 0
"GET /admin/auth/user/4/change/ HTTP/1.1" 200 22733
"GET /admin/jsi18n/ HTTP/1.1" 200 3342
"POST /login/ HTTP/1.1" 200 5780
"POST /login/ HTTP/1.1" 200 5780
"GET /login/ HTTP/1.1" 200 5416
"POST /login/ HTTP/1.1" 200 5780
"GET /signup/ HTTP/1.1" 200 7493
"POST /signup/ HTTP/1.1" 302 0
"GET / HTTP/1.1" 200 4214
"GET /rooms/ HTTP/1.1" 200 4311
"GET /rooms/Familly/ HTTP/1.1" 200 5428
"GET /rooms/Familly/ HTTP/1.1" 200 5428
"POST /rooms/Familly/ HTTP/1.1" 302 0
"GET /rooms/Familly/ HTTP/1.1" 200 5699
"GET /rooms/Familly/ HTTP/1.1" 200 5699
"GET /logout/ HTTP/1.1" 302 0
"GET / HTTP/1.1" 200 4123
"GET /logout/ HTTP/1.1" 302 0
"GET / HTTP/1.1" 200 4123
"GET /login/ HTTP/1.1" 200 5416
"POST /login/ HTTP/1.1" 302 0
"GET /rooms/ HTTP/1.1" 200 4311
"GET /login/ HTTP/1.1" 200 5766
"POST /login/ HTTP/1.1" 200 5780
"POST /login/ HTTP/1.1" 302 0
"GET /rooms/ HTTP/1.1" 200 4311
"GET /rooms/Familly/ HTTP/1.1" 200 5699
"GET /rooms/Familly/ HTTP/1.1" 200 5699
"POST /rooms/Familly/ HTTP/1.1" 302 0
"GET /rooms/Familly/ HTTP/1.1" 200 5970
"GET /rooms/Familly/ HTTP/1.1" 200 5970
"POST /rooms/Familly/ HTTP/1.1" 302 0
"GET /rooms/Familly/ HTTP/1.1" 200 6246
"GET /rooms/Familly/ HTTP/1.1" 200 6246
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 4214
Not Found: /favicon.ico
"GET /favicon.ico HTTP/1.1" 404 3376
"GET / HTTP/1.1" 200 4123
"GET /login/ HTTP/1.1" 200 5846
"POST /login/ HTTP/1.1" 302 0
"GET /rooms/ HTTP/1.1" 200 4311
"GET /login/ HTTP/1.1" 200 5416
"POST /login/ HTTP/1.1" 302 0
"GET /rooms/ HTTP/1.1" 200 4311
"GET /rooms/Familly/ HTTP/1.1" 200 7810
Not Found: /ws/chat/Familly/
"GET /ws/chat/Familly/ HTTP/1.1" 404 3391
- Broken pipe from ('127.0.0.1', 49310)
"GET /rooms/Familly/ HTTP/1.1" 200 7810
Not Found: /ws/chat/Familly/
"GET /ws/chat/Familly/ HTTP/1.1" 404 3391
"POST /rooms/Familly/ HTTP/1.1" 302 0
"GET /rooms/Familly/ HTTP/1.1" 200 8085
Not Found: /ws/chat/Familly/
"GET /ws/chat/Familly/ HTTP/1.1" 404 3391
"GET /rooms/Familly/ HTTP/1.1" 200 8085
Not Found: /ws/chat/Familly/
"GET /ws/chat/Familly/ HTTP/1.1" 404 3391
"GET /rooms/Familly/ HTTP/1.1" 200 8085
Not Found: /ws/chat/Familly/
"GET /ws/chat/Familly/ HTTP/1.1" 404 3391
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\urls.py changed, reloading.
Watching for file changes with StatReloader
"GET /rooms/Familly/ HTTP/1.1" 200 9438
"GET /rooms/Familly/ HTTP/1.1" 200 9438
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
"POST /rooms/Familly/ HTTP/1.1" 302 0
"GET /rooms/Familly/ HTTP/1.1" 200 9734
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
"POST /rooms/Familly/ HTTP/1.1" 302 0
"GET /rooms/Familly/ HTTP/1.1" 200 10030
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
"GET /logout/ HTTP/1.1" 302 0
"GET / HTTP/1.1" 200 4123
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=7 HTTP/1.1" 404 3412
"GET /logout/ HTTP/1.1" 302 0
"GET / HTTP/1.1" 200 4123
"GET /login/ HTTP/1.1" 200 5755
"POST /login/ HTTP/1.1" 302 0
"GET /rooms/ HTTP/1.1" 200 4311
"GET /login/ HTTP/1.1" 200 5758
"POST /login/ HTTP/1.1" 302 0
"GET /rooms/ HTTP/1.1" 200 4311
"GET /rooms/Familly/ HTTP/1.1" 200 10030
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=9 HTTP/1.1" 404 3412
"GET /rooms/Familly/ HTTP/1.1" 200 10030
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=9 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=9 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=9 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=9 HTTP/1.1" 404 3412
"POST /rooms/Familly/ HTTP/1.1" 302 0
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=9 HTTP/1.1" 404 3412
"GET /rooms/Familly/ HTTP/1.1" 200 10331
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=9 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=9 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=9 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=9 HTTP/1.1" 404 3412
"GET /rooms/Familly/ HTTP/1.1" 200 10331
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=9 HTTP/1.1" 404 3412
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
"GET /rooms/Familly/ HTTP/1.1" 200 10331
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\urls.py changed, reloading.
Watching for file changes with StatReloader
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Not Found: /Familly/messages/
"GET /Familly/messages/?last_message_id=10 HTTP/1.1" 404 3413
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 4214
"GET / HTTP/1.1" 200 4214
"GET /login/ HTTP/1.1" 200 5846
"POST /login/ HTTP/1.1" 302 0
"GET /rooms/ HTTP/1.1" 200 4311
"GET /rooms/ HTTP/1.1" 200 4311
Internal Server Error: /rooms/Familly/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 14, in room_detail
    room = get_object_or_404(ChatRoom, slug=slug)
           ^^^^^^^^^^^^^^^^^
NameError: name 'get_object_or_404' is not defined
"GET /rooms/Familly/ HTTP/1.1" 500 73926
Internal Server Error: /rooms/Familly/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 14, in room_detail
    room = get_object_or_404(ChatRoom, slug=slug)
           ^^^^^^^^^^^^^^^^^
NameError: name 'get_object_or_404' is not defined
"GET /rooms/Familly/ HTTP/1.1" 500 74189
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /rooms/Familly/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 26, in room_detail
    return render(request, 'core/room_detail.html', {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\template\loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
django.template.exceptions.TemplateDoesNotExist: core/room_detail.html
"GET /rooms/Familly/ HTTP/1.1" 500 91410
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py changed, reloading.
Watching for file changes with StatReloader
"GET /rooms/Familly/ HTTP/1.1" 200 10641
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18636
"GET /rooms/Familly/ HTTP/1.1" 200 10641
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 20, in room_detail
    return JsonResponse({
           ^^^^^^^^^^^^
NameError: name 'JsonResponse' is not defined
"POST /rooms/Familly/ HTTP/1.1" 500 75068
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 20, in room_detail
    return JsonResponse({
           ^^^^^^^^^^^^
NameError: name 'JsonResponse' is not defined
"POST /rooms/Familly/ HTTP/1.1" 500 75068
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 20, in room_detail
    return JsonResponse({
           ^^^^^^^^^^^^
NameError: name 'JsonResponse' is not defined
"POST /rooms/Familly/ HTTP/1.1" 500 75068
Internal Server Error: /rooms/Familly/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 20, in room_detail
    return JsonResponse({
           ^^^^^^^^^^^^
NameError: name 'JsonResponse' is not defined
"POST /rooms/Familly/ HTTP/1.1" 500 75068
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18636
"GET /logout/ HTTP/1.1" 302 0
"GET / HTTP/1.1" 200 4123
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=10 HTTP/1.1" 500 18487
"GET /logout/ HTTP/1.1" 302 0
"GET / HTTP/1.1" 200 4123
"GET /login/ HTTP/1.1" 200 5755
"POST /login/ HTTP/1.1" 302 0
"GET /rooms/ HTTP/1.1" 200 4311
"GET /login/ HTTP/1.1" 200 5758
"POST /login/ HTTP/1.1" 302 0
"GET /rooms/ HTTP/1.1" 200 4311
"GET /rooms/Familly/ HTTP/1.1" 200 11813
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18487
"GET /rooms/Familly/ HTTP/1.1" 200 11813
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 20, in room_detail
    return JsonResponse({
           ^^^^^^^^^^^^
NameError: name 'JsonResponse' is not defined
"POST /rooms/Familly/ HTTP/1.1" 500 75068
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18487
"GET /rooms/Familly/ HTTP/1.1" 200 12106
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=15 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=15 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=15 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=15 HTTP/1.1" 500 18636
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18487
Internal Server Error: /rooms/Familly/stream/
Traceback (most recent call last):
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 60, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\avira\Documents\Enfold_chat\Enfold_chat\room\views.py", line 58, in message_stream
    response = StreamingHttpResponse(event_stream(), content_type='text/event-stream')
               ^^^^^^^^^^^^^^^^^^^^^
NameError: name 'StreamingHttpResponse' is not defined
"GET /rooms/Familly/stream/?last_id=14 HTTP/1.1" 500 18487
Watching for file changes with StatReloader
"GET / HTTP/1.1" 200 4214
Not Found: /favicon.ico
"GET /favicon.ico HTTP/1.1" 404 3376
"GET /login/ HTTP/1.1" 200 5846
"POST /login/ HTTP/1.1" 302 0
"GET /rooms/ HTTP/1.1" 200 4311
"GET /rooms/Familly/ HTTP/1.1" 200 8673
"POST /rooms/Familly/ HTTP/1.1" 302 0
"GET /rooms/Familly/ HTTP/1.1" 200 8941
"GET /admin/ HTTP/1.1" 302 0
"GET /admin/login/?next=/admin/ HTTP/1.1" 200 3796
"POST /admin/login/?next=/admin/ HTTP/1.1" 302 0
"GET /admin/ HTTP/1.1" 200 7714
"GET /admin/room/chatroom/ HTTP/1.1" 200 9083
"GET /admin/jsi18n/ HTTP/1.1" 200 3342
"GET /admin/room/chatroom/1/change/ HTTP/1.1" 200 10101
"GET /admin/jsi18n/ HTTP/1.1" 200 3342
"GET /admin/room/chatroom/ HTTP/1.1" 200 9083
"GET /admin/room/chatroom/1/change/ HTTP/1.1" 200 10101
"GET /admin/jsi18n/ HTTP/1.1" 200 3342
"GET /admin/auth/group/ HTTP/1.1" 200 8129
"GET /admin/jsi18n/ HTTP/1.1" 200 3342
"GET /admin/auth/user/ HTTP/1.1" 200 14836
"GET /admin/jsi18n/ HTTP/1.1" 200 3342
"GET /admin/room/chatroom/ HTTP/1.1" 200 9083
"GET /admin/jsi18n/ HTTP/1.1" 200 3342
"GET /admin/room/chatroom/1/change/ HTTP/1.1" 200 10101
"GET /admin/jsi18n/ HTTP/1.1" 200 3342
"GET /admin/room/chatroom/1/history/ HTTP/1.1" 200 7035
"GET /admin/room/chatroom/1/change/ HTTP/1.1" 200 10101
"GET /admin/room/chatroom/ HTTP/1.1" 200 9083
"GET /admin/auth/user/ HTTP/1.1" 200 14836
"GET /admin/ HTTP/1.1" 200 7714
"GET /rooms/Familly/ HTTP/1.1" 200 8941
"GET /admin/auth/user/4/change/ HTTP/1.1" 200 22597
"GET /admin/jsi18n/ HTTP/1.1" 200 3342
"GET /static/admin/img/icon-unknown.svg HTTP/1.1" 200 655
"GET /static/admin/img/icon-unknown-alt.svg HTTP/1.1" 200 655
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\Enfold_chat\settings.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\Enfold_chat\settings.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\avira\Documents\Enfold_chat\Enfold_chat\Enfold_chat\settings.py changed, reloading.
Watching for file changes with StatReloader
